packages/core/src/renderer/components/+namespaces/namespace-tree-view.test.tsx

```diff
@@ -0,0 +1,264 @@
+});
+
+const teamA = createNamespace("team-a", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "acme-org.tree.hnc.x-k8s.io/depth": "1",
+  "kubernetes.io/metadata.name": "team-a",
+  "team-a.tree.hnc.x-k8s.io/depth": "0",
+});
+
+const teamB = createNamespace("team-b", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "acme-org.tree.hnc.x-k8s.io/depth": "1",
+  "kubernetes.io/metadata.name": "team-b",
+  "team-b.tree.hnc.x-k8s.io/depth": "0",
+});
+
+const teamC = createNamespace("team-c", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "org-a.tree.hnc.x-k8s.io/depth": "1",
+  "kubernetes.io/metadata.name": "team-c",
+  "team-c.tree.hnc.x-k8s.io/depth": "0",
+});
+
+const service1 = createNamespace("service-1", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "org-a.tree.hnc.x-k8s.io/depth": "1",
+  "kubernetes.io/metadata.name": "team-c",
+  "service-1.tree.hnc.x-k8s.io/depth": "0",
+}, {
+  "hnc.x-k8s.io/subnamespace-of": "org-a",
+});
+
+const levelsDeep = createNamespace("levels-deep", {
+  "hnc.x-k8s.io/included-namespace": "true",
+});
+
+const levelDeepChildA = createNamespace("level-deep-child-a", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "levels-deep.tree.hnc.x-k8s.io/depth": "1",
+  "level-deep-child-a.tree.hnc.x-k8s.io/depth": "0",
+});
+
+const levelDeepChildB = createNamespace("level-deep-child-b", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "levels-deep.tree.hnc.x-k8s.io/depth": "1",
+  "level-deep-child-b.tree.hnc.x-k8s.io/depth": "0",
+});
+
+const levelDeepSubChildA = createNamespace("level-deep-subchild-a", {
+  "hnc.x-k8s.io/included-namespace": "true",
+  "levels-deep.tree.hnc.x-k8s.io/depth": "2",
+  "level-deep-child-b.tree.hnc.x-k8s.io/depth": "1",
+  "level-deep-subchild-a.tree.hnc.x-k8s.io/depth": "0",
+});
+
+describe("<NamespaceTreeView />", () => {
+  let di: DiContainer;
+  let render: DiRender;
+  let namespaceStore: NamespaceStore;
+  let fetchMock: AsyncFnMock<Fetch>;
+  let cleanup: Disposer;
+  
+  beforeEach(async () => {
+    di = getDiForUnitTesting({ doGeneralOverrides: true });
+    di.unoverride(subscribeStoresInjectable);
+
+    di.override(directoryForUserDataInjectable, () => "/some-user-store-path");
+    di.override(directoryForKubeConfigsInjectable, () => "/some-kube-configs");
+    di.override(storesAndApisCanBeCreatedInjectable, () => true);
+
+    fetchMock = asyncFn();
+    di.override(fetchInjectable, () => fetchMock);
+
+    const createCluster = di.inject(createClusterInjectable);
+
+    di.override(hostedClusterInjectable, () => createCluster({
+      contextName: "some-context-name",
+      id: "some-cluster-id",
+      kubeConfigPath: "/some-path-to-a-kubeconfig",
+    }, {
+      clusterServerUrl: "https://localhost:8080",
+    }));
+
+    namespaceStore = di.inject(namespaceStoreInjectable);
+
+    const subscribeStores = di.inject(subscribeStoresInjectable);
+
+    cleanup = disposer(subscribeStores([namespaceStore]));
+    render = renderFor(di);
```

Comment:
Looks like `<NamespaceTreeView />` needs lot of things to setup for the test. Why? It should not require or even know about some stores imho and just render some sort of data-tree -> dom-tree representations. Or I miss something? 🤔